packages/palette/src/elements/Join/Join.tsx

```diff
@@ -26,15 +26,18 @@ interface JoinProps {
  */
 export const Join: SFC<JoinProps> = ({ separator, children }) => {
   const childArray = React.Children.toArray(children) as any
+  const filteredChildren = childArray.filter(
+    child => child && child.props && child.props.children
```

Comment:
What if it's just `<FooComponent />` without props or children?

Reply:
So i believe that is exactly the thing that we want to skip. If the component does not render anything whatsoever - it should be skipped. Like in out case <TestComponent/> returns `null`.

If the component renders anything - those elements considered `children` (or at least what it seems like from my testing). Similar with `<div/>` - blank one will be skipped, but once it has any content - it'll be considered with children.

Reply:
I wonder if you can assert against `child.type` vs checking against props and if that might be safer? 

Either way, an additional test case using a flat component (`<FooComponent />`) that _does_ have content will settle it.

Reply:
@damassi - I was wrong... It does not work if `<FooComponent />` just renders but does not have any params or children :(

Discussed this with @zephraph in person a bit - and even though we agree that the idea is nice here - it seems to be super tricky to implement without first rendering it.

I think I'll close this for now. Wondering though if it's helpful to just add the story that illustrates current functionality.

Reply:
Yup, we should keep the story since that now gets piped to Chromatic 👍